aten/src/ATen/cuda/CachingHostAllocator.h

```diff
@@ -18,25 +18,38 @@ namespace at::cuda {
 // call between host and device, and passed the corresponding context from the
 // allocation. This is currently invoked by at::native::copy_kernel_cuda.
 //
-TORCH_CUDA_CPP_API c10::Allocator* getCachingHostAllocator();
+inline TORCH_CUDA_CPP_API at::HostAllocator* getCachingHostAllocator() {
```

Comment:
Why do we call this a "caching" host allocator in these APIs rather that just `getHostAllocator`?
From what I can tell, all access to this just need a host allocator..

Do you think we could simplify the naming here by removing the "caching" concept from the user APIs ?

Reply:
The code change here is just for the backward compatibility. I plan to deprecate these device-specific APIs in this file in next [PR](https://github.com/pytorch/pytorch/pull/151437). And recommend the user to new unified APIs to cover this.